test/integration/allocatable_test.go

```diff
@@ -0,0 +1,194 @@
+/*
+Copyright 2020 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package integration
+
+import (
+	"context"
+	"testing"
+	"time"
+
+	v1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/api/resource"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/util/wait"
+	"k8s.io/kubernetes/pkg/scheduler"
+	schedapi "k8s.io/kubernetes/pkg/scheduler/apis/config"
+	fwkruntime "k8s.io/kubernetes/pkg/scheduler/framework/runtime"
+	st "k8s.io/kubernetes/pkg/scheduler/testing"
+	testutils "k8s.io/kubernetes/test/integration/util"
+	imageutils "k8s.io/kubernetes/test/utils/image"
+
+	"sigs.k8s.io/scheduler-plugins/pkg/noderesources"
+	"sigs.k8s.io/scheduler-plugins/test/util"
+)
+
+func TestAllocatablePlugin(t *testing.T) {
+	registry := fwkruntime.Registry{noderesources.AllocatableName: noderesources.NewAllocatable}
+	profile := schedapi.KubeSchedulerProfile{
+		SchedulerName: v1.DefaultSchedulerName,
+		Plugins: &schedapi.Plugins{
+			Score: &schedapi.PluginSet{
+				Enabled: []schedapi.Plugin{
+					{Name: noderesources.AllocatableName,
+						Weight: 50000},
+				},
+				Disabled: []schedapi.Plugin{
+					{Name: "*"},
+				},
+			},
+		},
+	}
+
+	testCtx := util.InitTestSchedulerWithOptions(
+		t,
+		testutils.InitTestMaster(t, "sched-allocatable", nil),
+		true,
+		scheduler.WithProfiles(profile),
+		scheduler.WithFrameworkOutOfTreeRegistry(registry),
+	)
+
+	defer testutils.CleanupTest(t, testCtx)
+
+	cs, ns := testCtx.ClientSet, testCtx.NS.Name
+	// Create a Node.
+	nodeName1 := "fake-node-small-1"
```

Comment:
We can put the node preparation logic into a for..loop.

Reply:
done